### PR TITLE
Updating vispy to 0.12.1

### PR DESCRIPTION
**Pull request by ryanskeith**
- Updated version and hash
- Some corrections for linting

### CODE REVIEW

**Comment by ryanskeith** [comment]
The linter error seems to be a bug.

**Review by varlackc — COMMENTED** [review]
It might be recommended to have a skip section where we set the python pinnings:
https://github.com/vispy/vispy/blob/v0.12.1/setup.py#L95

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
 requirements:
   build:
+    - {{ compiler('c') }}
     - patch     # [not win]
     - m2-patch  # [win]
   host:
     - pip
     - python
-    - setuptools
+    - setuptools >=30.3.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:31]: setuptools and cython should be without pinnings, https://github.com/vispy/vispy/blob/7accf231d251271eae6edfa3e1b103c15bc0a95e/setup.py#L97

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,33 +1,40 @@
...
+  sha256: e00881c1d0f9b2c08eb6e26e9361adbde8a3a96e7878ee6c1e199e7c5872224f
   patches:
     - ctypes_fontconfig.diff
 
 build:
   number: 0
   script: {{ PYTHON }} -m pip install . -vv
+  ignore_run_exports:
+    - fontconfig
+    - python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:21]: Why do we ignore run exports of fontconfig and python? What will be if we remove `ignore_run_exports`?

File: recipe/meta.yaml (hunk 1 above)
Comment (ryanskeith) [reply recipe/meta.yaml:31]: I got the requirement from here: https://github.com/vispy/vispy/blob/7accf231d251271eae6edfa3e1b103c15bc0a95e/pyproject.toml#L4
and here:
https://github.com/vispy/vispy/blob/7accf231d251271eae6edfa3e1b103c15bc0a95e/pyproject.toml#L10

File: recipe/meta.yaml (hunk 2 above)
Comment (ryanskeith) [reply recipe/meta.yaml:21]: Interestingly enough, I get an overdepending error if these are not included.

**Pull request merged**